A/ADOLC/build_tarballs.jl

```diff
@@ -0,0 +1,97 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+# See https://discourse.julialang.org/t/binarybuilder-jl-cant-dlopen-because-of-libopenlibm-so/108486
+# This build script needs to be run with julia 1.7
+
+# See https://github.com/JuliaLang/Pkg.jl/issues/2942
+# Once this Pkg issue is resolved, this must be removed
+uuid = Base.UUID("a83860b7-747b-57cf-bf1f-3e79990d037f")
+delete!(Pkg.Types.get_last_stdlibs(v"1.6.3"), uuid)
+
+name = "ADOLC"
+version = v"0.0.24"
+
+isyggdrasil = get(ENV, "YGGDRASIL", "") == "true"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/coin-or/ADOL-C.git", "b7ea10536a1e452f7dd47a90ef8c8118ce3e8432"),
+    GitSource("https://github.com/TimSiebert1/libadolccxx.git", "51c23abf9ee2e1566331b440e81503ac9e7e8f5b"),
+]
+
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd ADOL-C/ && autoreconf -fi && ./configure --enable-atrig-erf --prefix=${prefix} --build=${MACHTYPE} --host=${target}
+make -j${nproc}
+make install
+mkdir ${prefix}/share/licenses/ADOLC
+install_license LICENSE
+
+cd ../libadolccxx/
+cmake -DCMAKE_INSTALL_PREFIX=${prefix}\
+      -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN}\
+      -DCMAKE_BUILD_TYPE=Release\
+      -DADOLC_DIR=${WORKSPACE}/destdir\
+      -DJulia_PREFIX=${prefix}\
+      -B build .
+
+cd build
+make -j${nprocs}
+make install
+"""
+
+# These are the platforms we will build for by default, unless further
+# platforms are passed in on the command line
+
+platforms = [
+    Platform("x86_64", "linux"; libc = "glibc"),
+    Platform("aarch64", "macos"; ),
+    Platform("x86_64", "windows"; )
+    # Platform("i686", "linux"; libc = "glibc"),
+    # Platform("aarch64", "linux"; libc = "glibc"),
+    # Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    # Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    # Platform("powerpc64le", "linux"; libc = "glibc"),
+    # Platform("x86_64", "linux"; libc = "musl"),
+    # Platform("x86_64", "macos"; ),
+    # Platform("x86_64", "freebsd"; ),
+    # Platform("i686", "windows"; ),
+]
+
+# See https://discourse.julialang.org/t/binarybuilder-jl-cant-dlopen-because-of-libopenlibm-so/108486
+# It seems we need a separate build for each Julia version
+if isyggdrasil
+    include("../../L/libjulia/common.jl")
+else
+    # For local test, download
+    # https://github.com/JuliaPackaging/Yggdrasil/blob/master/L/libjulia/common.jl
+    # https://github.com/JuliaPackaging/Yggdrasil/blob/master/fancy_toys.jl
+    include("common.jl")
+end
+julia_versions=VersionNumber[v"1.9.0", v"1.10.0"]
```

Comment:
Why this restriction? If there is a proper reason, it should be stated. Otherwise, better to get rid of it to maximize where this can be used (e.g. right now you are excluding the 1.11 betas/release candidates).
```suggestion
```

Reply:
See above. We will have to test with 1.11 before modifying this.

Reply:
I don't see why you'd have to test? Unless you directly call into libjulia yourself, this should just work. Even in the worst case (i.e., it does not work after all), you can then deal with this by adjusting your Julia package which is using this JLL to either not load it there (if the JLL is for an optional feature), or to refuse to load itself (if the JLL is essential). But again, it usually works everywhere if it works on one Julia version -- except when there is a breaking change in the Julia API/ABI and you have the bad luck of trying to build the JLL before we had a chance to adjust CxxWrap. But that's not currently the case as far as I know.

Hence, I'd really suggest to not artificially limit the list of supported Julia versions here unless you really have explicit knowledge of a failure (i.e. I suggest to assume it works by default, while you seem to assume that "it doesn't work" as default)

Reply:
Without 
```
julia_versions=VersionNumber[v"1.9", v"1.10"]
platforms = vcat(libjulia_platforms.(julia_versions)...)
```
`libjulia.so.1` cannot be loaded and the wrapper package tests fail. 

Also, for 1.11 I am getting compilation errors, probably due to the new Memory based array layout.
Once @TimSiebert1 merges https://github.com/TimSiebert1/Yggdrasil/pull/1 , the corresponding change will occur here.

Reply:
Huh? If you don't specify `julia_versions` then all that does is that the default value for it is used, which is a super set of what you use. So your statement that then libjulia cannot be loaded is odd. It is not clear to me what you mean by that either. Who cannot load it? What is the error?

Reply:
Ahh wait you also removed `platforms = vcat(libjulia_platforms.(julia_versions)...)` but that must stay.